tests/unit/api/c/dmx_merger/test_dmx_merger.cpp

```diff
@@ -210,9 +101,104 @@ class TestDmxMerger : public ::testing::Test
   sacn_source_id_t slot_owners_[DMX_ADDRESS_COUNT];
   sacn_dmx_merger_t merger_handle_;
   SacnDmxMergerConfig merger_config_;
+};
+
+/* This fixture is for testing the update (merging) functions of the merger API. */
+class TestDmxMergerUpdate : public TestDmxMerger
+{
+protected:
+  void SetUp() override
+  {
+    TestDmxMerger::SetUp();
+
+    // Initialize the merger and sources.
+    EXPECT_EQ(sacn_dmx_merger_create(&merger_config_, &merger_handle_), kEtcPalErrOk);
+    EXPECT_EQ(sacn_dmx_merger_add_source(merger_handle_, &merge_source_1_), kEtcPalErrOk);
+    EXPECT_EQ(sacn_dmx_merger_add_source(merger_handle_, &merge_source_2_), kEtcPalErrOk);
 
-  uint8_t test_values_ascending_[DMX_ADDRESS_COUNT];
-  uint8_t test_values_descending_[DMX_ADDRESS_COUNT];
+    // Initialize expected merge results.
+    ClearExpectedMergeResults();
+
+    // Initialize test values to be used for levels and PAPs.
+    test_values_ascending_.resize(DMX_ADDRESS_COUNT);
+    test_values_descending_.resize(DMX_ADDRESS_COUNT);
+
+    for (int i = 0; i < DMX_ADDRESS_COUNT; ++i)
```

Comment:
The problem is that then I wouldn't be able to initialize them with the for loops (the Intellisense complains when I put the loops in the constructor, requiring a modifiable l-value). So then I would need {1, 2, 3, 4, 5, ...} for up to 512 values. Unless if that is still preferable or there's a different way to do this?

Reply:
There's a technique I've seen a few places that essentially just does the initialization inside a lambda so that you can do things like for loops while keeping the created variable constant. That could be handy here.
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-lambda-init

Reply:
Very cool @jtveite ! I didn't know lambdas could be used that way. It looks like this works. Thanks!

Reply:
Implemented in https://github.com/ETCLabs/sACN/pull/29/commits/93551f5ed28d503c1f06b4d067a021256f788372

Reply:
Cool trick @jtveite , I am gonna start using this in my code